packages/interfaces/src/service/navigation-service.interface.ts

```diff
@@ -56,6 +56,28 @@ export interface Navigator {
      * @return {Location}
      */
     getLocation(): Location;
+
+    /**
+     * Add a blocker that can prevent navigation away from the current page, for
+     * example when there is some unsaved data that would otherwise be discarded.
+     *
+     * If the blocker function returns true navigation is blocked. If it returns
+     * false navigation will proceed as normal.
+     *
+     * A location is provided to the blocker function. This could be used to later
+     * on resume the navigation, for instance as the result of some user response or
+     * equivalent.
```

Comment:
I think this piece of information is more helpful if it's turned around like, "To implement [functionality], you can use this thing" rather than "There is this thing. You can use is it to implement [functionality]."

(also, "the equivalent of some user response" is just wordiness to me)

```suggestion
     * To later on resume navigation that was blocked, for instance as the result of some user response,
     * you may save the new location provided to the blocker function.
```

... but now that I think about it, how's the blocker function supposed to save the blocked location? Maybe it's more helpful if the navigator service could provide the most recently blocked location instead?

Reply:
Now, I haven't looked at the whole PR, so it's likely I'm missing stuff here, but:

It sounds a little odd to me that the blocking-function would be involved in resuming navigation… 🤔

Is the plan to let each blocking-function display its own warning dialog to the user? Unless we're sure we'll never need to have more than one blocking function registered at any one time, it seems to me like we should rather let some central thing control that. But I might be biased by our old solution… 🤷‍♂️ 😅

Reply:
Just realised my suggestion made this ambiguity worse:

"navigation that was blocked, [...] as the result of some user response"

Reply:
I added your suggestion for improvement to the doctoring in a fixup commit. Could you please have a look?

Reply:
> At first I made the blocking functions async with the notion that the blocker should return true or false based on user input. But after talking to @jgroth it's clear that you shouldn't wait for something

Having them return a `Promise<boolean>` could be a valid use case, but I'm not sure how everything should work then :thinking: What would happen after all promises have been resolved? Should we retry the transition when all of them resolved to `false` perhaps? :thinking: 
Some of the old routes in the Angular router have these "Guards" and those could return a whole bunch of different things, `Promise<boolean>` being one of them

https://github.com/Lundalogik/lime-webclient/blob/main/frontend/webclient/src/app/guards/ngu-admin-guard.service.ts#L10

Having these return a promise seems to complicate the use case a bit, but we can always add it later if we think it seems like a good idea :slightly_smiling_face: